### DISCUSSION TITLE
Toggletip

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#298] [Component suggestions]
<img width="480" alt="visuele weergave van de toggletip component" src="https://github.com/user-attachments/assets/0c8130f4-e40d-4d80-8dd1-3532bbf44678">

We willen graag meer leren over de **Toggletip** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1730289605203889)

---

## Naam
Toggletip

## Doel
Een knop die tijdelijke informatie toont en deze automatisch verbergt wanneer iets anders dan de knop wordt gebruikt. 

## Ook bekend als
Tooltip

## Onderzoek
[Tooltips & Toggletips - Inclusive Components](https://inclusive-components.design/tooltips-toggletips/)
In dit artikel wordt besproken wanneer je een Toggletip moet gebruiken en hoe je deze toegankelijk en inclusief kunt implementeren. De auteur stelt dat Toggletips vaak onnodig complex zijn voor informatie die net zo goed direct in de tekst zou kunnen staan. 

[Tooltips and Toggletips - Pearson](https://arc.net/l/quote/wtfivovs)
Overzicht van toegankelijkheidseisen voor een Toggletip. 

[Toggletip gebruikersonderzoek - RVO](https://github.com/orgs/nl-design-system/discussions/298#discussioncomment-11119942)
In de onderstaande reactie geeft Robert aan dat uit gebruikersonderzoek naar de Toggletip blijkt dat het ‘i’-icoon niet door gebruikers werd begrepen.

[Target Size (Minimum) (Level AA) - W3C](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html) 
Er wordt aanbevolen dat klikbare elementen, zoals iconen, een minimale grootte van 24 x 24 px hebben.

**Comment by hidde** [comment]
Ik vind de naam verwarrend bij het doel en het screenshot, die lezen/ogen meer als een tooltip? 

Wat een toggletip uniek maakt is denk ik dat het expliciet door een actie van de gebruiker wordt getoggled, bv met het klikken op een knop, terwijl een tooltip uit zichzelf verschijnt en verdwijnt, bv door hover of focus.

**Comment by Rozerinay** [comment]
## DSO

- [Documentatie website](https://www.dso-toolkit.nl/66.1.0/components/on-page-interactives/toggletip)
- [Storybook](https://storybook.dso-toolkit.nl/66.1.0/?path=/docs/core-toggletip--docs)
- [GitHub](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/toggletip)


---


### Screenshots / Varianten


**Storybook**

Info button
<img width="44" alt="image" src="https://github.com/user-attachments/assets/31277c24-e97d-4312-a30c-a3810d2ac98f">

Position right
<img width="640" alt="image" src="https://github.com/user-attachments/assets/5172be55-8f43-4fb7-91a3-0357fe964c90">

Position top
Note: Inhoud van Toggletip wordt eronder gepositioneerd vanwege de beperkte ruimte aan de bovenzijde.
<img width="640" alt="image" src="https://github.com/user-attachments/assets/5d57084d-4ead-40c9-9000-70272c896c74">

Position bottom
<img width="640" alt="image" src="https://github.com/user-attachments/assets/5bc5eb10-062d-44a8-84ad-e2f048799226">

Position left
Note: Inhoud van Toggletip is niet goed zichtbaar vanwege de beperkte ruimte aan de linkerzijde.
<img width="68" alt="image" src="https://github.com/user-attachments/assets/70ca7a02-2f28-4d35-a968-18ca096aebc5">

Small
<img width="375" alt="image" src="https://github.com/user-attachments/assets/3d6f6e87-17a6-436b-ac51-5e910dafdbf4">

Secondary
<img width="51" alt="image" src="https://github.com/user-attachments/assets/297c791e-359e-4dd5-a2a6-a7cb0395b839">

<img width="640" alt="image" src="https://github.com/user-attachments/assets/e687f50e-b161-48de-a488-9c1d45cc1303">

Responsive gedrag
<img width="300" alt="image" src="https://github.com/user-attachments/assets/1e9bca8c-0ecf-4493-a9d3-40602fb3eac5">

---


### Notities
- Info button heeft default en secondary variant.
- Hover: andere styling. 
- Positionering van de inhoud van Toggletip is instelbaar: top, right, bottom, left. 
- Keuze in small inhoud: wel of niet. 
- Toegankelijk label is instelbaar en is standaard 'Toelichting' als deze niet is gedefinieerd.
- Toggletip sluit bij een klik overal op de pagina. 
- Inhoud van Toggletip kan bestaan uit een Heading, Paragraph en Link.
- Richtlijnen beschikbaar.

**Comment by Rozerinay** [comment]
## RVO

[Reactie in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1730362385086959?thread_ts=1730289605.203889&cid=C7KDQRXMZ)

---

### Screenshots / Varianten

<image src="https://github.com/user-attachments/assets/8260d665-ff7d-4b9f-87c6-58af848f42ff" width="800px"/>

---

### Toelichting Robert Roose

Wij zijn afgestapt van dit component omdat klanten het 'i'-tje niet begrepen en dat alleen overheidsmedewerkers het snapte. Ik kan dit helaas niet onderbouwen met harde onderzoeksdata maar wellicht goed om dit component nog eens te laten testen.

In het voorbeeld hierboven is een formulier in een zaaksysteem te zien dat is getest.

**Comment by MarjonBakker** [comment]
Zie ook het artikel [Tooltips in the time of WCAG 2.1](https://sarahmhigley.com/writing/tooltips-in-wcag-21/) van Sarah Higley. Al wat ouder, 2019, maar staat nog steeds.

**Comment by jeffreylauwers** [comment]
Zie ook het mini-artikel [Simple Standalone Toggletip widget pattern](https://www.tpgi.com/simple-standalone-toggletip-widget-pattern/) van Steve Faulkner. Hij doet een voorzet voor een stukje code om dit component op te maken. Er wordt gebruik gemaakt van JS. Ik kan niet beoordelen of dat goed is.

**Comment by rianrietveld** [comment]
@jeffreylauwers @Rozerinay 
Mijn mening: Deze component is moeilijk toegankelijk te maken (screenreader, en vooral inzoomen). Essentiele informatie moet nooit achter een klein knopje zitten, zeker als het hele lappen tekst zijn.
Beter is om de informatie boven het formulier of in de description te zetten, of het formulier goed te overdenken zodat zoveel extra informatie om iets uit te leggen niet nodig is bij een invoerveld. Liever een meerstappen formulier dan bij alles knopjes met uitleg.
Ik zie het als een last resort, maar liever helemaal niet.

Ik kan me voorstellen dat het toch moet omdat anders er ontoegankelijke versies gaan rondzwerven, maar echt, ik vind dit _bad UX_ en een zwaktebod om info door te geven.

**Reply by MarjonBakker** [reply]
Helemaal mee eens, @rianrietveld .

**Reply by Yolijn** [reply]
We verwachten dat het inderdaad belangrijk is een toegankelijke oplossing voor dit component te vinden. Maar ook dat we hier wel naartoe willen werken door de juiste documentatie en voorbeelden te verzamelen. Daarom maken we dit component nu wel Help Wanted.

**Comment by Yolijn** [comment]
## ✨ Dit component is nu Help Wanted ✨


Help je mee hem door de Community stappen te krijgen?
[Toggletip op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=170)
